dataset/src/test/scala/frameless/FilterTests.scala

```diff
@@ -144,4 +144,12 @@ class FilterTests extends TypedDatasetSuite {
     check(forAll(prop[Option[X1[X1[String]]]] _))
     check(forAll(prop[Option[X1[X1[Vector[Option[Int]]]]]] _))
   }
+
+  test("filter with isin values") {
+    check(forAll { (data: Vector[X1[Int]], values: Vector[Int]) =>
+      val ds = TypedDataset.create(data)
+      val res = ds.filter(ds('a).isin(values:_*)).collect().run().toVector
+      res ?= data.filter(d => values.contains(d.a))
+    })
```

Comment:
Could you parametrize this test on the `Int`? It would be interesting to see how this works for other types (`SQLDate`, `String`, `X1[String]`, ...)

Reply:
Good point! will do it this evening

Reply:
Surprisingly if fails for `SQLDate` . we get `Cause: java.lang.RuntimeException: Unsupported literal type class frameless.SQLDate SQLDate`. I guess we should limit `isin` to primitive types ? Or @OlivierBlanvillain do you have an idea on handling case classes ?

Reply:
Sounds like it. What about other used defined case classes, such as `X1[String]`, `X2[String, Int]`,?

Reply:
Same issue since SQLDate is also a case class:
```
Cause: java.lang.RuntimeException: Unsupported literal type class frameless.X1 X1(Ȗ沭䁎Ꞛ蟽啮琏銉綿鋪鎸㭝ꜰﮤ菝˔赘䑕⏐៼ꖃ댬ự뷴뒎ﺱ㟯천䪼汹题켭)
[info]   at org.apache.spark.sql.catalyst.expressions.Literal$.apply(literals.scala:77)
[info]   at org.apache.spark.sql.catalyst.expressions.Literal$$anonfun$create$2.apply(literals.scala:163)
[info]   at org.apache.spark.sql.catalyst.expressions.Literal$$anonfun$create$2.apply(literals.scala:163)
[info]   at scala.util.Try.getOrElse(Try.scala:79)
[info]   at org.apache.spark.sql.catalyst.expressions.Literal$.create(literals.scala:162)
[info]   at org.apache.spark.sql.functions$.typedLit(functions.scala:112)
[info]   at org.apache.spark.sql.functions$.lit(functions.scala:95)
[info]   at org.apache.spark.sql.Column$$anonfun$isin$1.apply(Column.scala:779)
[info]   at org.apache.spark.sql.Column$$anonfun$isin$1.apply(Column.scala:779)
[info]   at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
[info]   at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
[info]   at scala.collection.Iterator$class.foreach(Iterator.scala:891)
[info]   at scala.collection.AbstractIterator.foreach(Iterator.scala:1334)
[info]   at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
[info]   at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
[info]   at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
[info]   at scala.collection.AbstractTraversable.map(Traversable.scala:104)
[info]   at org.apache.spark.sql.Column.isin(Column.scala:779)
```

it boils down to spark calling [typedLit](https://github.com/apache/spark/blob/v2.2.1/sql/core/src/main/scala/org/apache/spark/sql/functions.scala#L109) and since the value isn't a `Column` or a `Symbol`, spark tries to create a literal for it which fails [here](https://github.com/apache/spark/blob/v2.2.1/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/literals.scala#L77)

@OlivierBlanvillain Is it because frameless serialize literals differently than spark ?